DBZ-3960 Provide list of DBZ UI configurations (#248)
debezium/docker-images: README.md

Debezium is open source under the [Apache License, Version 2.0](http://www.apache.org/licenses/LICENSE-2.0.html)

# Purpose of the Debezium UI
The Debezium UI provides standalone web application, which connects to Kafka Connect via its REST API. See the [Debezium UI docs](https://debezium.io/documentation/reference/operations/debezium-ui.html) for more information

# Debezium environment vars
The table shows the ENV vars for the [Debezium UI container image](https://hub.docker.com/r/debezium/debezium-ui) and it's related `application.properties` parameter name when running the java application locally.

| ENV vars        | application.properties vars  | Default value             | Description |
| :-------------- |:-----------------------------|:--------------------------|:------------|
| DEPLOYMENT_MODE | deployment.mode              | default                   | Parameter to enable/disable connector properties validation mode in the Debezium UI, to disable the validation change value to `validation.disabled` |

> | KAFKA_CONNECT_URI|kafka.connect.uri            | http://connect:8083       | Provide existing Kafka connection instance url. `KAFKA_CONNECT_URI` supplies the comma-separated list of available URI(s) |

Reviewer: @uidoyen that looks already quite good. Maybe you can take a look into the Debezium connectors documentation and see how we document the connector configuration in such tables there. 
We normally list the available options with a brief description. Maybe we can do this here as well?

Author: @rk3rn3r updated the vars description accordingly.

Debezium is open source under the [Apache License, Version 2.0](http://www.apache.org/licenses/LICENSE-2.0.html)

# Purpose of the Debezium UI
The Debezium UI provides standalone web application, which connects to Kafka Connect via its REST API. See the [Debezium UI docs](https://debezium.io/documentation/reference/operations/debezium-ui.html) for more information

> # Debezium environment vars

Reviewer: Could you add a small sentence explaining the table? Saying that it shows the ENV vars for the Debezium UI container image (with Docker Hub link) and it's related application.properties parameter name when running the java application locally.

Reviewer: ```suggestion
# Configure the Debezium UI
```

| ENV vars        | application.properties vars  | description |
| :-------------- |:-----------------------------|:------------|
| UI_BASE_URI     | ui.base.uri                  | http://localhost:8080/api |
| UI_MODE         | ui.mode                      | production |

> | DEPLOYMENT_MODE | deployment.mode              | validation.disabled |

Reviewer: `KAFKA_CONNECT_URI` is missing

> 

Reviewer: Could you add a real explanation for the config properties and maybe move the default values into an extra column?

Debezium is open source under the [Apache License, Version 2.0](http://www.apache.org/licenses/LICENSE-2.0.html)

# Purpose of the Debezium UI
The Debezium UI provides standalone web application, which connects to Kafka Connect via its REST API. See the [Debezium UI docs](https://debezium.io/documentation/reference/operations/debezium-ui.html) for more information

# Debezium environment vars

> The table shows the ENV vars for the [Debezium UI container image](https://hub.docker.com/r/debezium/debezium-ui) and it's related `application.properties` parameter name when running the java application locally.

Reviewer: ```suggestion
The following table shows the environment variables for the [Debezium UI container image](https://hub.docker.com/r/debezium/debezium-ui) and the related parameter names inside `application.properties` when running the Java application without the container.
```

<table id="debezium">
  <thead>
    <tr>

>       <th>ENV vars</th>

Reviewer: ```suggestion
      <th>Environment variable</th>
```

>       <th>application.properties vars</th>

Reviewer: ```suggestion
      <th>Parameter name in application.properties</th>
```

      <th>Default value</th>
      <th>Description</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td valign="top">DEPLOYMENT_MODE</td>
      <td valign="top">deployment.mode</td>
      <td valign="top">default</td>

>       <td valign="top">Specifies how connector properties validation mode is enabled/disabled when configuring the connectors in the Debezium UI <br><br><code>default</code>: When set to default it turns the validation mode on for the frontend<br><br> <code>validation.disabled</code>:  When set to validation.disabled it turns the validation mode off for the frontend<br>

Reviewer: ```suggestion
      <td valign="top">Specifies how the Debezium UI is deployed.<br>For example, in some environments it might not be possible to reach the underlying backend, Kafka Connect REST interface or databases, then the deployment mode can be switched to match the underlying infrastructure.
      <br><br><code>default</code>: The default deployment mode. It uses the Debezium UI backend with the configured Kafka Connect clusters via the Kafka Connect REST interface (see <a href="#KAFKA_CONNECT_URI">KAFKA_CONNECT_URI</a> how they are configured).
      <br><br> <code>validation.disabled</code>: When set to validation.disabled the UI frontend will not call the backend to validate the user input nor check the availability and proper configuration of database connections. That mode is used to only generate the Debezium connector JSON configuration without the UI backend validation.<br>
```

      </td>
    </tr>
    <tr>

>         <td valign="top">KAFKA_CONNECT_URI</td>

Reviewer: ```suggestion
        <td valign="top"><a id="KAFKA_CONNECT_URI">KAFKA_CONNECT_URI</a></td>
```

Reviewer: Could you also add links to the different paramters on the other env vars

        <td valign="top">kafka.connect.uri</td>
        <td valign="top">http://connect:8083</td>

>         <td valign="top">Supplies the comma-separated list of available Kafka Connect clusters URI(s) and provides a REST interface for the frontend.</td>

Reviewer: ```suggestion
        <td valign="top">A comma-separated list to one or more URLs of Kafka Connect REST interfaces to specify the Kafka Connect clusters that should be managed by the Debezium UI.</td>
```

      <br><br> <code>validation.disabled</code>: When set to validation.disabled the UI frontend will not call the backend to validate the user input nor check the availability and proper configuration of database connections. That mode is used to only generate the Debezium connector JSON configuration without the UI backend validation.<br>
      </td>
    </tr>
    <tr>
        <td valign="top"><a id="KAFKA_CONNECT_URI">KAFKA_CONNECT_URI</a></td>
        <td valign="top">kafka.connect.uri</td>
        <td valign="top">http://connect:8083</td>
        <td valign="top">A comma-separated list to one or more URLs of Kafka Connect REST interfaces to specify the Kafka Connect clusters that should be managed by the Debezium UI.</td>
    </tr>        
  </tbody>

> </table>

Reviewer: `UI_MODE` is not used anymore?
